### PR TITLE
Generate pyproject dependencies with dfg

### PR DESCRIPTION
**Pull request by vyasr**
This PR uses dependencies.yaml to generate the dependency lists in pyproject.toml

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

File: dependencies.yaml [hunk 1]
```diff
@@ -21,7 +24,7 @@ files:
     includes:
       - cudatoolkit
       - py_version
-      - test_python
+      - test_python_common
```
Comment (vyasr) [inline dependencies.yaml:27]:
```suggestion
      - test_python_common
      - test_python_pylibraft
```

File: dependencies.yaml [hunk 2]
```diff
@@ -21,7 +24,8 @@ files:
...
+      - test_python_pylibraft
```
Comment (vyasr) [inline dependencies.yaml:28]:
```suggestion
      - test_pylibraft
```

File: python/raft-dask/pyproject.toml [hunk 3]
```diff
@@ -34,15 +34,15 @@ authors = [
 license = { text = "Apache 2.0" }
 requires-python = ">=3.8"
 dependencies = [
-    "numpy",
-    "numba>=0.49",
-    "joblib>=0.11",
     "dask-cuda==23.4.*",
     "dask>=2023.1.1",
-    "ucx-py==0.31.*",
```
Comment (cjnolet) [inline python/raft-dask/pyproject.toml:42]: Why does ucx-py get removed here?

**Review by cjnolet — APPROVED** [review]
Just a small question but otherwise it looks great. Pre-approving.

**Pull request merged**